tools/c7n_sphinxext/c7n_sphinxext/docgen.py

```diff
@@ -176,16 +159,40 @@ def _main(provider, output_dir, group_by):
 
     files = []
 
-    # Write out resources by grouped page
-    for key, group in itertools.groupby(
-            sorted(provider_class.resources.values(), key=group_by), key=group_by):
-        rpath = os.path.join(output_dir, "%s.rst" % key)
+    groups = {}
+
+    for r in provider_class.resources.values():
+        group = group_by(r)
+        if type(group) != list:
+            group = [group]
```

Comment:
nit this should probably be a set for the temp accumulator, also is instance is better equality check on types